transcendence/api.py

```diff
@@ -23,27 +24,46 @@
 # delete the jwt token from the cookie
 
 def logout(request):
-    response = HttpResponseRedirect('/')
-    response.delete_cookie('jwt')
-    return response
+    pass
 
 def guest_login(request):
-    guest_uuid = shortuuid.ShortUUID().random(length=22)
-    avatar = user.assign_random_avatar()
-    color = user.assign_random_background_color()
+    guest_id = shortuuid.ShortUUID().random(length=22)
+    now = int(time.time())
+    
+    payload = {
+        'id': guest_id,
+        'guest': True,
+        'iat': now,
+        'exp': now + 3600,
+    }
+
+    jwt_token = jwt.encode(payload, settings.JWT_SECRET_KEY, algorithm=settings.JWT_ALGORITHM)
+
+    avatar = assign_random_avatar()
+    color = assign_random_background_color()
     # save_user_cache(guest_id, avatar, color, guest=True)
-    return JsonResponse({'guest_login': guest_uuid})
+
+    # redirect to the main page with jwt token as cookie set
+    response = HttpResponseRedirect('/')  # Redirect to the dashboard or desired URL
+    response.set_cookie(
+        key='jwt', 
+        value=jwt_token, 
+        httponly=True,  # make the cookie inaccessible to js
+        secure=True,  # ensure the cookie is sent over HTTPS
+        samesite='Lax'  # define when to allow the cookie to be sent
+    )
+    return response
 
 # check if user is authenticated
 def check_auth(request):
-    jwt_token = request.COOKIES.get('jwt')
+    jwt_token = request.cookies.get('jwt')
     try:
         payload = jwt.decode(jwt_token, settings.JWT_SECRET_KEY, algorithms=settings.JWT_ALGORITHM)
-    except jwt.InvalidTokenError:
-        return JsonResponse({'error': 'Invalid token'})
+    except jwt.invalidTokenError:
+        return {'error': 'Invalid token'}
     except jwt.ExpiredSignatureError:
-        return JsonResponse({'error': 'Token has expired'})
-    return JsonResponse(payload)
```

Comment:
I also changed that. It has to return JsonResponse and not Json because this function is called in url.py and is expecting a JsonResponse